### PR TITLE
GoogleApiClient is in App class now. 

### PR DESCRIPTION
**Pull request by tasomaniac**
Google API Client is created in every Activity. Which is I think bad.
Since we are using it almost everywhere, I think it is better to define it once in Application class.

What do you think? If anyone have time, can you test it?

### CODE REVIEW

File: app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java [hunk 1]
```diff
@@ -100,38 +89,24 @@ protected String getTrackedViewName() {
...
-                    .addApi(AppStateManager.API)
-                    .addScope(Plus.SCOPE_PLUS_LOGIN)
-                    .addScope(Plus.SCOPE_PLUS_PROFILE)
-                    .addScope(Games.SCOPE_GAMES)
-                    .build();
-        }
-
-        mAchievementActionHandler =
-                new AchievementActionHandler(getHandler(), mGoogleApiClient, this);
+        getGoogleApiClient().registerConnectionFailedListener(this);
```
Comment (friedger) [inline app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java]: This is leaking the activity, unregistration is necessary.

File: app/src/main/java/org/gdg/frisbee/android/common/GdgNavDrawerActivity.java [hunk 2]
```diff
@@ -77,6 +78,13 @@
     private MenuItem drawerItemToNavigateAfterSignIn = null;
     private static final int GROUP_ID = 1;
 
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        getGoogleApiClient().registerConnectionCallbacks(this);
```
Comment (friedger) [inline app/src/main/java/org/gdg/frisbee/android/common/GdgNavDrawerActivity.java]: This is leaking the activity, unregistration is necessary.

**Comment by friedger** [comment]
The documentation says to instantiate in the activity onCreate method.

**Comment by friedger** [comment]
From http://stackoverflow.com/questions/16827839/access-google-plus-client-from-multiple-activities

>Each Activity should maintain its own instance of GoogleApiClient since it is a lightweight object used to access the shared state held by the Google Play services process. 

So, I'd keep it like it is now.

**Comment by friedger** [comment]
Maybe we could use enableAutoMange https://developers.google.com/android/reference/com/google/android/gms/common/api/GoogleApiClient.Builder#enableAutoManage(android.support.v4.app.FragmentActivity, int, com.google.android.gms.common.api.GoogleApiClient.OnConnectionFailedListener)

File: app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java (hunk 1 above)
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java]: Do you know what is the difference between doin it like this or using `addConnectionCallbacks` in the builder. When we do it in the builder we never unregister.

**Comment by tasomaniac** [comment]
You are right. But I've been seeing lot of leakage because of GoogleApiClient already. The problem is that when you navigate through Nav Drawer we are opening every activity on top of each other. After 10 nav drawer clicks, we have to press back 10 times. This means 10 GoogleApiClient object being allocated. Instead of removing GoogleApiClient we can fix that issue.

**Pull request closed without merging**